### PR TITLE
Add the ability for modlogs to send to a channel

### PR DESCRIPTION
**Pull request by Minion3665**
- Change the config format from .json to .ts (allowing easier typing & doc)
- Update templates.ActionCommand.ts to new configs
- Finish updating utils to new configs
- Fix rule numbering
- Make logging work
- Make logging correctly specify the moderator
- Previously a command would let you send a notification to someone
  higher than you saying that you had moderated them before failing to
  take the action for kicks and bans. This has been fixed
- Prevent pinging in the moderation log message to stop multiple pings
  for multi-channel logs and 'at everyone in reason' vulnerabilities

Undefined behavior:
- If the message is too long, this doesn't paginate it, and if the
  message fails to send the moderation action will ignore it and run
  anyway. This means that it may be possible to run a moderation action
  without logging it in the channels

Deployment caveats:
- As the rules format changes in this PR, data migration will be
  required
- As the config format changes in this PR, config migration will be
  required

### CODE REVIEW

**Comment by PineaFan** [comment]
All changes look good :+1: 
("Reviewed" for @Minion3665 )

**Comment by Minion3665** [comment]
> All changes look good +1 ("Reviewed" for @Minion3665 )

Sounds good, thank you Ash

@Cleover do you want to take another look on call w/ me before I merge or no?

File: src/utils.ts [hunk 1]
```diff
@@ -1,61 +1,90 @@
...
  */
 function getConfigPath(name: string, ext: string) {
-  const DEV_CONF = path.join(SRC_PATH, `../configs/${name}.dev.${ext}`);
-  const PROD_CONF = path.join(SRC_PATH, `../configs/${name}.${ext}`);
+  if (!ext.startsWith('.')) ext = `.${ext}`;
+
+  let dev_conf;
+  let prod_conf;
+  if (ext === '.ts') {
+    dev_conf = path.join(SRC_PATH, `../dist/configs/${name}.dev${ext}`);
```
Comment (Minion3665) [inline src/utils.ts:29]: This should be .js rather than ${ext}

File: configs/rulesType.d.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export interface ruleType {
```
Comment (Minion3665) [inline configs/rulesType.d.ts:1]: can we add a short ID string for storage in mongo?

File: src/resources/commandTemplates/templates.ActionCommand.ts [hunk 3]
```diff
@@ -25,6 +27,61 @@ function getBasicOptions(interaction: CommandInteraction) {
...
+async function formatLogMessage(client: Client, user: User, log: ModerationLog, extraActionOptions: ExtraActionOptions, _removePrivateInfo = false): Promise<string> {
+  let moderator: User | null = null;
+  try {
+    moderator = await client.users.fetch(log.moderator);
+  } catch (e) {
+    if (!(e instanceof DiscordAPIError)) throw e;
+  }
+
+  return `${extraActionOptions.emoji} ${moderator ?? 'Unknown'} *${extraActionOptions.pastTense}* ${log.userState.username}#${log.userState.discriminator} [\`${user.id}\`, <@${user.id}>]\n` +
+         `> ${log.reason}` +
```
Comment (Minion3665) [inline src/resources/commandTemplates/templates.ActionCommand.ts:51]: Remove undefined behavior if reason length is too long

**Review by Cleover — COMMENTED** [review]
Resolve the above before merging

**Review by Cleover — APPROVED** [review]
Looks good 👍

**Pull request merged**